Intersect.Client/Interface/Game/EntityPanel/EntityBox.cs

```diff
@@ -593,158 +628,216 @@ private void UpdateHpBar(float elapsedTime, bool instant = false)
             else
             {
                 HpLbl.Text = Strings.EntityBox.vital0val.ToString(0, 0);
-                targetHpWidth = HpBackground.Width;
+                targetHpSize = HpBackground.Width;
+
+                if (DirectionVitalsRender == "vertical")
+                {
+                    targetHpSize = HpBackground.Height;
+                }
             }
 
-            if ((int)targetHpWidth != CurHpWidth)
+            if ((int)targetHpSize != CurHpSize)
             {
                 if (!instant)
                 {
-                    if ((int)targetHpWidth > CurHpWidth)
+                    if ((int)targetHpSize > CurHpSize)
                     {
-                        CurHpWidth += 100f * elapsedTime;
-                        if (CurHpWidth > (int)targetHpWidth)
+                        CurHpSize += 100f * elapsedTime;
+                        if (CurHpSize > (int)targetHpSize)
                         {
-                            CurHpWidth = targetHpWidth;
+                            CurHpSize = targetHpSize;
                         }
                     }
                     else
                     {
-                        CurHpWidth -= 100f * elapsedTime;
-                        if (CurHpWidth < targetHpWidth)
+                        CurHpSize -= 100f * elapsedTime;
+                        if (CurHpSize < targetHpSize)
                         {
-                            CurHpWidth = targetHpWidth;
+                            CurHpSize = targetHpSize;
                         }
                     }
                 }
                 else
                 {
-                    CurHpWidth = (int)targetHpWidth;
+                    CurHpSize = (int)targetHpSize;
                 }
 
-                if (CurHpWidth == 0)
+                if (CurHpSize == 0)
                 {
                     HpBar.IsHidden = true;
                 }
                 else
                 {
-                    HpBar.Width = (int)CurHpWidth;
-                    HpBar.SetTextureRect(0, 0, (int)CurHpWidth, HpBar.Height);
+                    if(DirectionVitalsRender == "horizontal")
+                    {
+                        HpBar.Width = (int)CurHpSize;
+                        HpBar.SetTextureRect(0, 0, (int)CurHpSize, HpBar.Height);
+                    }
+                    else
+                    {
+                        var backgroundFactor = HpBackground.Height - (int)CurHpSize;
+                        HpBar.SetBounds(HpBar.X, HpBackground.Y + backgroundFactor, HpBar.Width, (int)CurHpSize);
+                        HpBar.SetTextureRect(0, backgroundFactor, HpBar.Width, (int)CurHpSize);
+                    }
```

Comment:
Extract into reusable method for all bars, passing in the foreground image panel, the background image panel, the direction, and the current size.

Also the actual changing of bounds/texturerect on the bar should be done once, and the values that are passed in should be computed in a switch (since on discord we discussed 4 different render directions).